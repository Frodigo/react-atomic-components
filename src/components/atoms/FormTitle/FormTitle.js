import React from 'react';
import PropTypes from 'prop-types';

import classes from './FormTitle.module.css';

/**
 * This is the FormTitle.
 * Component that display form label/legend
 * @param {*} props
 */
const FormTitle = (props) => {
    const { type, children, ...rest } = props;

    if (type === 'legend') {
        return (
            <legend className={classes.legend} {...rest}>
                {children}
            </legend>
        );
    }

    return (
        <label className={classes.label} {...rest}>
            {children}
        </label>
    );
};

FormTitle.propTypes = {
    htmlFor: PropTypes.string,
    type: PropTypes.oneOf(['label', 'legend'])
};

FormTitle.defaultProps = {
    type: 'label'
};

export default FormTitle;
