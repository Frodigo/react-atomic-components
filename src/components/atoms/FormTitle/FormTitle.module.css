.label {
    color: var(--color-text);
    font-size: 16px;
    font-weight: var(--font-weight-bold);
    line-height: 1.75;
    letter-spacing: normal;
    margin-bottom: var(--spacing-xs);
}

.legend {
    border-bottom: 1px solid var(--color-primary);
    color: var(--color-primary);
    display: block;
    font-size: 18px;
    font-weight: 300;
    line-height: normal;
    padding-bottom: 12px;
}
