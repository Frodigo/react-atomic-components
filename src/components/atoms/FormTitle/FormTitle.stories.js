import React from 'react';

import { default as FormTitle } from './FormTitle';

export default {
    title: 'Atoms/Form/FormTitle',
    component: FormTitle,
    argTypes: {}
};

const Template = (args) => <FormTitle {...args} />;

export const Label = Template.bind({});
Label.args = {
    htmlFor: 'inputId',
    children: 'This is the label'
};

export const Legend = Template.bind({});
Legend.args = {
    type: 'legend',
    children: 'This is the legend'
};
