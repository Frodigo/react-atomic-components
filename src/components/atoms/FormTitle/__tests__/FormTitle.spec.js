import React from 'react';
import { render } from '@testing-library/react';

import FormTitle from '../FormTitle';

test('Label renders correctly', () => {
    const text = 'This is the label';
    const { container } = render(<FormTitle>{text}</FormTitle>);

    expect(container).toMatchSnapshot();
});

test('Label has for attribute', () => {
    const text = 'This is the label';
    const { container } = render(
        <FormTitle htmlFor="inputId">{text}</FormTitle>
    );

    expect(container.firstChild.getAttribute('for')).toStrictEqual('inputId');
});

test('Legend renders correctly', () => {
    const text = 'This is the legend';
    const { container } = render(<FormTitle type="legend">{text}</FormTitle>);

    expect(container).toMatchSnapshot();
});
