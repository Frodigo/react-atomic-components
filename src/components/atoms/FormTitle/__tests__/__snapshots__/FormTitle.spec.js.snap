// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label renders correctly 1`] = `
<div>
  <label
    class="label"
  >
    This is the label
  </label>
</div>
`;

exports[`Legend renders correctly 1`] = `
<div>
  <legend
    class="legend"
  >
    This is the legend
  </legend>
</div>
`;
